grails-app/services/org/pih/warehouse/inventory/CycleCountService.groovy

```diff
@@ -64,6 +64,12 @@ class CycleCountService {
             if (command.sort) {
                 getCandidatesSortOrder(command.sort, command.order, delegate, usedAliases)
             }
+            if (command.status) {
+                createAlias("cycleCountRequest", "cycleCountRequest", JoinType.INNER_JOIN)
+                eq("cycleCountRequest.status", command.status)
+                return
+            }
+            isNull("cycleCountRequest")
```

Comment:
e.g. 
1. for "to count" tab we will filter by `CREATED` status
2. for all products tab we don't provide status, so we want to search for candidates that do not have the cycle count request created

Reply:
We also need to filter out requests that are currently in progress. Those shouldn't show up on the all products tab.

Reply:
on all products tab we don't provide the status property, hence we would return all candidates that DO NOT have the cycle count request.

Reply:
and they won't show, because the cycle count request would already exist for such candidate, so those would be filtered out.

Reply:
is that true? Maybe I'm misunderstanding, but for all products tab we don't provide a status and so we do `isNull("cycleCountRequest")` and fetch only the rows that don't have a request.

I don't want to go in circles about this so if you tested it and it's working, I trust you and we can move on :laughing:

Reply:
We discussed on slack. I'm being silly. This code works.